hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -162,8 +163,11 @@ protected void doRefresh() {
       Thread.currentThread().interrupt();
       throw new RuntimeException("Interrupted during refresh", e);
     }
-
-    refreshFromMetadataLocation(metadataLocation, metadataRefreshMaxRetries);
+    if (table != null && table.getTableType().equalsIgnoreCase(TableType.VIRTUAL_VIEW.name())) {
+      disableRefresh();
```

Comment:
Why is this?

Reply:
If requested table is of View type we should not refresh. we have same for other catalog like [InMemory](https://github.com/apache/iceberg/blob/f246614ed9d98f84eae13ff57768d377a95d3ced/core/src/main/java/org/apache/iceberg/inmemory/InMemoryCatalog.java#L377-L384).

Reply:
I am curious, do you know why we do not refresh?

Reply:
Should we just throw an error immediately then?

Reply:
Since we are using `ViewCatalogTests` error should be also consistent with test cases.
We have different errors for different scenarios. 
 IMO it is better to have the consistency with other catalogs. 
WDYT ?

Reply:
What would it take to throw the correct exception here? How do we end up the same exception, if we avoiding refresh?

Reply:
If we are avoiding refresh and throwing any exception here, one of the followings can be done:

1. Add the same errors to the corresponding tests something similar was tried at previous [commits](https://github.com/apache/iceberg/commit/477c7d1c24526e2983462c8fdd1f418264441aac#diff-9c8b69f86aec86aea82cd6675c1c24267d9a9d77d7715c2c792934cb954b328bR371). 
2. Or Implement the same tests for HiveCatalog with its own error.
3. Or make all the Catalog consistent with error handling for such cases.

Reply:
What is the exception that is thrown here using the `InMemoryCatalog`?
Is it the correct behaviour for the `InMemoryCatalog`?